aten/src/ATen/SparseTensorUtils.h

```diff
@@ -58,6 +58,17 @@ inline Tensor new_values_with_size_of(const Tensor& values, int64_t nnz) {
   return at::empty(size, values.options());
 }
 
+inline Tensor new_values_with_size_of(const Tensor& values, int64_t nnz, ScalarType dtype) {
+  std::vector<int64_t> size = values.sizes().vec();
+  size[0] = nnz;
+  return at::empty(size, values.options().dtype(dtype));
+}
+
+inline Tensor maybe_promoted_tensor(const Tensor & tensor, ScalarType commonDtype) {
+  const Tensor& promoted = tensor.scalar_type() != commonDtype? tensor.to(commonDtype) : tensor;
```

Comment:
Isn't this just `tensor.to(commonDtype)`? If the tensor already has the correct dtype the operation doesn't do anything iirc

Reply:
hmm... I think I made an incorrect assumption here based on something I thought came out of the tensor iterator changes. Looking at the implementation, you're right I don't see it doing anything for matching dtypes.

Reply:
removed this